sormas-backend/src/main/java/de/symeda/sormas/backend/caze/CaseService.java

```diff
@@ -876,4 +904,120 @@ private Predicate createNewCaseFilter(CriteriaBuilder cb, From<?, Case> caze, Da
 
 		return newCaseFilter;
 	}
+
+	public Set<Case> getAllRelevantCases(Person person, Disease disease, Date referenceDate) {
+
+		CriteriaBuilder cb = em.getCriteriaBuilder();
+		CriteriaQuery<Case> cq = cb.createQuery(Case.class);
+		Root<Case> caseRoot = cq.from(Case.class);
+
+		Predicate filter =
+				and(cb, createDefaultFilter(cb, caseRoot), buildRelevantCasesFilter(person, disease, referenceDate, cb, caseRoot));
+		cq.where(filter);
+
+		return new HashSet<>(em.createQuery(cq).getResultList());
+	}
+
+	/**
+	 * Returns a filter that can be used to retrieve all cases with the specified
+	 * person and disease whose report date is before the reference date and,
+	 * if available, whose follow-up  until date is after the reference date,
+	 * including an offset to allow some tolerance.
+	 */
+	private Predicate buildRelevantCasesFilter(Person person, Disease disease, Date referenceDate, CriteriaBuilder cb, Root<Case> from) {
```

Comment:
Give this a more specific name, e.g. buildRelevantCasesFilterForFollowUp

Reply:
Ok